core/cache/lxdprofilewatcher.go

```diff
@@ -98,20 +98,21 @@ func (w *MachineAppLXDProfileWatcher) applicationCharmURLChange(topic string, va
 			w.logError(fmt.Sprintf("error getting charm %s to evaluate for lxd profile notification: %s", chURL, err))
 			return
 		}
+
 		// notify if:
 		// 1. the prior charm had a profile and the new one does not.
 		// 2. the new profile is not empty.
-		if (info.charmProfile != nil && ch.details.LXDProfile.Empty()) ||
-			!ch.details.LXDProfile.Empty() {
+		lxdProfile := ch.LXDProfile()
+		if (!info.charmProfile.Empty() && lxdProfile.Empty()) || !lxdProfile.Empty() {
 			logger.Tracef("notifying due to change of charm lxd profile for %s, machine-%s", appName, w.machineId)
 			notify = true
 		} else {
 			logger.Tracef("no notification of charm lxd profile needed for %s, machine-%s", appName, w.machineId)
 		}
-		if ch.details.LXDProfile.Empty() {
-			info.charmProfile = nil
+		if lxdProfile.Empty() {
```

Comment:
if ch.LXDProfile() no longer returns a pointer... is this check still necessary?  Just assign at line 115 all the time.

Reply:
Done.